File: C1.UWP.FlexChart/CS/SunburstIntro/ViewModel/SunburstViewModel.cs
using C1.Chart;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using Windows.UI.Xaml.Data;

namespace SunburstIntro
{
    public class SunburstViewModel : INotifyPropertyChanged
    {
        private string _legendPosition = "Auto";
        private string _selectedItemPosition = "Top";

        public List<DataItem> HierarchicalData
        {
            get
            {
                return DataService.CreateHierarchicalData();
            }
        }

        public List<FlatDataItem> FlatData
        {
            get
            {
                return DataService.CreateFlatData();
            }
        }

        public ICollectionView View
        {
            get
            {
                return DataService.CreateGroupCVData();
            }
        }

        public List<string> Positions
        {
            get
            {
                return Enum.GetNames(typeof(Position)).ToList();
            }
        }

        public List<string> Palettes
        {
            get
            {
                return Enum.GetNames(typeof(Palette)).ToList();
            }
        }

        public List<string> LabelPositions
        {
            get
            {
                return Enum.GetNames(typeof(PieLabelPosition)).ToList();
            }
        }

        public List<string> LabelOverlappings
        {
            get
            {
                return Enum.GetNames(typeof(PieLabelOverlapping)).ToList();
            }
        }

        public string LegendPosition
        {
            get { return _legendPosition; }
            set
            {
                if (_legendPosition != value)
                {
                    _legendPosition = value;
                    OnPropertyChanged("LegendPosition");
                }
            }
        }

        public string SelectedItemPosition
        {
            get { return _selectedItemPosition; }
            set
            {
                if (_selectedItemPosition != value)
                {
                    _selectedItemPosition = value;
                    OnPropertyChanged("SelectedItemPosition");
                }
            }
        }

        private void OnPropertyChanged(string propertyName)
        {
            if (PropertyChanged != null)
            {
                PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
            }
        }

        public event PropertyChangedEventHandler PropertyChanged;
    }
}

File: C1.UWP.FlexChart/VB/SunburstIntro/ViewModel/SunburstViewModel.vb
Imports System.Linq
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System
Imports C1.Chart

Public Class SunburstViewModel
    Implements INotifyPropertyChanged
    Private _legendPosition As String = "Auto"
    Private _selectedItemPosition As String = "Top"

    Public ReadOnly Property HierarchicalData() As List(Of DataItem)
        Get
            Return DataService.CreateHierarchicalData()
        End Get
    End Property

    Public ReadOnly Property FlatData() As List(Of FlatDataItem)
        Get
            Return DataService.CreateFlatData()
        End Get
    End Property

    Public ReadOnly Property View() As ICollectionView
        Get
            Return DataService.CreateGroupCVData()
        End Get
    End Property

    Public ReadOnly Property Positions() As List(Of String)
        Get
            Return [Enum].GetNames(GetType(Position)).ToList()
        End Get
    End Property

    Public ReadOnly Property Palettes() As List(Of String)
        Get
            Return [Enum].GetNames(GetType(Palette)).ToList()
        End Get
    End Property

    Public Property LegendPosition() As String
        Get
            Return _legendPosition
        End Get
        Set
            If _legendPosition <> Value Then
                _legendPosition = Value
                OnPropertyChanged("LegendPosition")
            End If
        End Set
    End Property

    Public Property SelectedItemPosition() As String
        Get
            Return _selectedItemPosition
        End Get
        Set
            If _selectedItemPosition <> Value Then
                _selectedItemPosition = Value
                OnPropertyChanged("SelectedItemPosition")
            End If
        End Set
    End Property

    Private Sub OnPropertyChanged(propertyName As String)
        RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
    End Sub

    Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged
End Class